Define repository-tag-based releases for the Go client (#537)
kubernetes-sigs/agent-sandbox: RELEASE.md

# Release Process

`agent-sandbox` uses repository-level semantic version tags (for example, `v0.1.0`).
Those tags are the source of truth for:

- Controller manifests published on GitHub Releases
- The Go SDK at `sigs.k8s.io/agent-sandbox/clients/go/sandbox`
- The Python SDK workflows that are triggered by `v*` tags

## Repository Release Flow

The project is released on an as-needed basis. The current process is:

> 1. Run `make release-promote TAG=vX.Y.Z` to create the repository tag, wait for the tagged image to be pushed, and generate the image promotion PR. Creating the Git tag also triggers the Python SDK release workflow.

Reviewer: Since pushing the `vX.Y.Z` tag also triggers the new `Release Go Client` workflow (defined in `.github/workflows/go-client-release.yml`), let's mention both workflows here for completeness.

```suggestion
1. Run `make release-promote TAG=vX.Y.Z` to create the repository tag, wait for the tagged image to be pushed, and generate the image promotion PR. Creating the Git tag also triggers the Python SDK and Go SDK release workflows. 
```

1. All [OWNERS](OWNERS) must LGTM the release.
1. An OWNER promotes images and prepares release assets as needed.

> 1. An OWNER pushes the repository tag for the release.

Reviewer: We don't actually follow these steps (RELEASE.md is generated from a repo template). 

Please see the release process I documented here: https://github.com/kubernetes-sigs/agent-sandbox/issues/408#issue-4073000607

We're working on automating that in a GitHub workflow.

Author: Thanks for the clarification. Updated RELEASE.md to reflect the current process documented in #408 instead of the repo-template release steps, and noted that these steps are being automated in GitHub Actions.

1. GitHub Actions validates the Go SDK and publishes or updates the draft GitHub Release assets.
1. The release issue is closed.

> 1. An announcement email is sent to `dev@kubernetes.io` with the subject `[ANNOUNCE] agent-sandbox $VERSION is released`.

Reviewer: ```suggestion
1. An announcement email is sent to `dev@kubernetes.io` and `sig-apps@kubernetes.io` with the subject `[ANNOUNCE] agent-sandbox $VERSION is released`.
```

> These steps are being automated in GitHub Actions so that a release only requires adding a repository tag.

Reviewer: ```suggestion
These steps are being automated in [GitHub Actions](https://github.com/kubernetes-sigs/agent-sandbox/blob/main/.github/workflows/release.yml) so that a release only requires adding a repository tag (occurs automatically on a schedule).
```

> ## Go SDK Releases

Reviewer: Please remove this part, and delete all other files related to `make release-go-sdk` and `dev/tools/release-go`. Client-only Git tagging cannot be safely bypassed around `make release-promote` in a single-module Go monorepo. It doesn't allow post releases like Python SDK. We will just couple Go SDK releases with the controller releases.

Author: Thanks for the clarification. That makes sense.

I’ll update the PR to align the Go client release process with the repository/controller release flow:

- The Go client version should come from the repository-level SemVer tag.
- Go client releases should happen together with controller releases.
- We should not support client-only Go SDK post releases.
- We should not provide a separate `release-go-sdk` entry point.

I’ll remove the standalone Go SDK release flow and keep the documentation focused on the coupled release model.
